### PR TITLE
Added Strings for keys

### PR DESCRIPTION
**Pull request by Nicolasdejean**
replaced byte * by String for the keys #104

### CODE REVIEW

File: src/TheThingsNetwork.h [hunk 1]
```diff
@@ -48,7 +48,7 @@ class TheThingsNetwork
     String readValue(String key);
     bool sendCommand(String cmd);
     bool sendCommand(String cmd, String value);
-    bool sendCommand(String cmd, const byte* buf, int length);
+    bool sendCommand(String cmd, const byte *buf, int length);
```
Comment (FokkeZB) [inline src/TheThingsNetwork.h]: Uhm.. space should follow `*` right?

File: src/TheThingsNetwork.h [hunk 2]
```diff
@@ -58,10 +58,10 @@ class TheThingsNetwork
     TheThingsNetwork(Stream& modemStream, Stream& debugStream, ttn_fp_t fp, int sf = TTN_DEFAULT_SF, int fsb = TTN_DEFAULT_FSB);
     void showStatus();
     void onMessage(void (*cb)(const byte* payload, int length, int port));
-    bool provision(const byte appEui[8], const byte appKey[16]);
-    bool join(const byte appEui[8], const byte appKey[16], int retries = -1, long int retryDelay = 10000);
+    bool provision(const String appEui, const String appKey);
```
Comment (FokkeZB) [inline src/TheThingsNetwork.h]: Tweetonig uses `const char* ` instead of `const String`... Strings are bad right?

File: src/TheThingsNetwork.cpp [hunk 3]
```diff
@@ -44,11 +44,12 @@ bool TheThingsNetwork::sendCommand(String cmd) {
 }
 
 bool TheThingsNetwork::sendCommand(String cmd, String value) {
-  int l = value.length();
-  byte buf[l];
-  value.getBytes(buf, l);
+  String str = "";
```
Comment (FokkeZB) [inline src/TheThingsNetwork.cpp]: Is this change related?

File: examples/Workshop/Workshop.ino [hunk 4]
```diff
@@ -6,16 +6,16 @@
 // After you registered your ABP device, go to The Things Network Dashboard
 // and copy the Device Addr, Network Session Key and App Session Key
 
-// Set your Device Address, for example: { 0x02, 0xDE, 0xAE, 0x00 };
-const byte devAddr[4] = { ... };
+// Set your Device Address, for example: "1C55B133";
+const String devAddr = " ... ";
```
Comment (FokkeZB) [inline examples/Workshop/Workshop.ino]: While you're at it, just change these constant placeholders to be like the other examples

File: src/TheThingsNetwork.cpp (hunk 3 above)
Comment (Nicolasdejean) [inline src/TheThingsNetwork.cpp]: I figured that we are sending Strings now and we didn't need to convert them in bytes, I wasn't sure about this... Should I create another sendComand() ?

Comment (FokkeZB) [inline src/TheThingsNetwork.cpp]: Well, as it is before this PR it uses the still existing `TheThingsNetwork::sendCommand(String cmd, const byte *buf, int length)`, not this `TheThingsNetwork::sendCommand(String cmd, String value)` right?

File: src/TheThingsNetwork.h (hunk 2 above)
Comment (johanstokking) [inline src/TheThingsNetwork.h]: Yep, `const char *`

File: src/TheThingsNetwork.cpp [hunk 5]
```diff
@@ -44,12 +44,11 @@ bool TheThingsNetwork::sendCommand(String cmd) {
...
-  String str = "";
-  str.concat(cmd);
-  str.concat(" ");
-  str.concat(value);
+  int l = value.length();
+  byte buf[l];
+  value.getBytes(buf, l);
 
-  return sendCommand(str);
+  return sendCommand(cmd, buf, l);
```
Comment (johanstokking) [inline src/TheThingsNetwork.cpp]: We don't need to convert back to buffer

File: src/TheThingsNetwork.cpp [hunk 6]
```diff
@@ -101,11 +100,18 @@ void TheThingsNetwork::onMessage(void (*cb)(const byte* payload, int length, int
...
+  String addr = "mac set devaddr ";
+  String nKey = "mac set nwkskey ";
+  String aKey = "mac set appskey ";
+
+  addr.concat(devAddr);
+  nKey.concat(nwkSKey);
+  aKey.concat(appSKey);
+  sendCommand(addr);
+  sendCommand(nKey);
+  sendCommand(aKey);
```
Comment (johanstokking) [inline src/TheThingsNetwork.cpp]: Keep the variables together, not the operations

File: src/TheThingsNetwork.cpp [hunk 7]
```diff
@@ -125,9 +131,14 @@ bool TheThingsNetwork::personalize() {
...
-  sendCommand(F("mac set appeui"), appEui);
-  sendCommand(F("mac set appkey"), appKey);
+bool TheThingsNetwork::provision(const char *appEui, const char *appKey) {
+  String eui = "mac set appeui ";
+  String key = "mac set appkey ";
+
+  key.concat(appKey);
+  eui.concat(appEui);
+  sendCommand(eui);
+  sendCommand(key);
```
Comment (johanstokking) [inline src/TheThingsNetwork.cpp]: Idem

File: src/TheThingsNetwork.cpp (hunk 6 above)
Comment (Nicolasdejean) [inline src/TheThingsNetwork.cpp]: I'm sorry but I am not sure I understand what you mean... Is it a spacing problem like that: 
```
String eui = "mac set appeui ";
String key = "mac set appkey ";
key.concat(appKey);
eui.concat(appEui);                                                                                         

sendCommand(eui);
sendCommand(key);
return sendCommand(F("mac save")); 
 ``` 
or something else ?
